src/devices_models/devices/AC_branches.jl

```diff
@@ -34,113 +36,55 @@ function flow_variables!(
 end
 
 #################################### Flow Variable Bounds ##################################################
-function branch_rate_bounds!(
-    psi_container::PSIContainer,
+function _get_constraint_data(
     devices::IS.FlattenIteratorWrapper{B},
-    ::Type{<:AbstractBranchFormulation},
-    ::Type{<:PM.AbstractDCPModel},
 ) where {B <: PSY.ACBranch}
-    constraint_data = Vector{DeviceRange}()
-    for d in devices
+    constraint_data = Vector{DeviceRange}(undef, length(devices))
+    for (ix, d) in enumerate(devices)
         limit_values = (min = -1 * PSY.get_rate(d), max = PSY.get_rate(d))
         name = PSY.get_name(d)
         services_ub = Vector{Symbol}()
         for service in PSY.get_services(d)
             SR = typeof(service)
             push!(services_ub, Symbol("R$(PSY.get_name(service))_$SR"))
         end
-        push!(
-            constraint_data,
-            DeviceRange(name, limit_values, services_ub, Vector{Symbol}()),
-        )
+        constraint_data[ix] = DeviceRange(name, limit_values, services_ub, Vector{Symbol}())
     end
-    set_variable_bounds!(psi_container, constraint_data, FLOW_ACTIVE_POWER, B)
-    return
+    return constraint_data
 end
 
 function branch_rate_bounds!(
     psi_container::PSIContainer,
     devices::IS.FlattenIteratorWrapper{B},
-    ::Type{D},
-    ::Type{S},
-) where {B <: PSY.ACBranch, D <: AbstractBranchFormulation, S <: PM.AbstractPowerModel}
-    constraint_data = Vector{DeviceRange}()
-
-    for d in devices
-        limit_values = (min = -1 * PSY.get_rate(d), max = PSY.get_rate(d))
-        name = PSY.get_name(d)
-        services_ub = Vector{Symbol}()
-        for service in PSY.get_services(d)
-            SR = typeof(service)
-            push!(services_ub, Symbol("R$(PSY.get_name(service))_$SR"))
-        end
-        push!(
-            constraint_data,
-            DeviceRange(name, limit_values, services_ub, Vector{Symbol}()),
-        )
-    end
-    set_variable_bounds!(psi_container, constraint_data, FLOW_ACTIVE_POWER_FROM_TO, B)
-    set_variable_bounds!(psi_container, constraint_data, FLOW_ACTIVE_POWER_TO_FROM, B)
+    model::DeviceModel{B, <:AbstractBranchFormulation},
+    ::Type{<:PM.AbstractDCPModel},
+) where {B <: PSY.ACBranch}
+    constraint_data = _get_constraint_data(devices)
+    set_variable_bounds!(psi_container, constraint_data, FLOW_ACTIVE_POWER, B)
```

Comment:
We don't need the directional limits anymore?

Reply:
The DCPModels have only one flow variable